share/smack/include/pthread.h

```diff
@@ -0,0 +1,71 @@
+//
```

Comment:
My guess would be that DSA nodes — and only those corresponding to pthread API data — would collapse only if the client program didn't treat those types opaquely, e.g., trying to access into `pthread_t` with its own (possibly incorrect) knowledge of the types. As far as I am aware, DSA attempts to infer node structure from the actual program accesses. And I am supposing that client programs do treat those types like `pthread_t` opaquely.

Anyways, I guess it shouldn’t be too hard to try this out. And if it allowed us to avoid that delicate front-end header stuff, perhaps it is worth it, no?